include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -600,6 +600,66 @@ using __repacked_tuple_t = typename __repacked_tuple<T>::type;
 template <typename _ContainerOrIterable>
 using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
 
+//-----------------------------------------------------------------------
+// future and helper classes for async pattern/algorithm
+//-----------------------------------------------------------------------
+
+struct __tmp_base
+{
+};
+
+template <typename... Ts>
+struct __TempObjs : public __tmp_base
+{
+    ::std::tuple<Ts&...> __my_tmps;
+    __TempObjs(Ts&... __t) : __my_tmps(::std::forward_as_tuple(__t...)) {}
```

Comment:
Why do we use `forward_as_tuple` when we even don't have anything to forward there?